presto-main/src/main/java/com/facebook/presto/operator/aggregation/ArrayAggregation.java

```diff
@@ -0,0 +1,171 @@
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(ArrayAggregation.class, "combine", Type.class, ArrayAggregationState.class, ArrayAggregationState.class);
+    private static final Signature SIGNATURE = new Signature(NAME, ImmutableList.of(typeParameter("T")), "array<T>", ImmutableList.of("T"), false, false);
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "return an array of values";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type type = types.get("T");
+        Signature signature = new Signature(NAME,
+                parameterizedTypeName("array", type.getTypeSignature()),
+                type.getTypeSignature());
+        InternalAggregationFunction aggregation = generateAggregation(type);
+        return new FunctionInfo(signature, getDescription(), aggregation);
+    }
+
+    protected InternalAggregationFunction generateAggregation(Type type)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(ArrayAggregation.class.getClassLoader());
+
+        AccumulatorStateSerializer<?> stateSerializer = new ArrayAggregationStateSerializer(type);
+        AccumulatorStateFactory<?> stateFactory = new ArrayAggregationStateFactory();
+        MethodHandle inputFunction = INPUT_FUNCTION.bindTo(type);
+        MethodHandle outputFunction = OUTPUT_FUNCTION.bindTo(type);
+        MethodHandle combineFunction = COMBINE_FUNCTION.bindTo(type);
+        Class<? extends AccumulatorState> stateInterface = ArrayAggregationState.class;
+
+        List<Type> inputTypes = ImmutableList.of(type);
+        Type outputType = new ArrayType(type);
+        Type intermediateType = stateSerializer.getSerializedType();
+        List<ParameterMetadata> inputParameterMetadata = createInputParameterMetadata(type);
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, type, inputTypes),
+                inputParameterMetadata,
+                inputFunction,
+                null,
+                null,
+                combineFunction,
+                outputFunction,
+                stateInterface,
+                stateSerializer,
+                stateFactory,
+                outputType,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, outputType, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type value)
+    {
+        return ImmutableList.of(new ParameterMetadata(STATE), new ParameterMetadata(NULLABLE_INPUT_CHANNEL, value), new ParameterMetadata(BLOCK_INDEX));
```

Comment:
Just noticed that this is also the default for all aggregate functions in presto. https://prestodb.io/docs/current/functions/aggregate.html